src/topology/mod.rs

```diff
@@ -225,6 +225,13 @@ impl RunningTopology {
             return Ok(false);
         }
 
+        // Synchronize source identifiers into the new config
+        for (name, source) in self.config.sources.iter() {
+            if let Some(new_source) = new_config.sources.get_mut(name) {
+                new_source.identifier = source.identifier.clone();
```

Comment:
Whether this causes issues will depend on how these identifiers get used, but we'll need to keep in mind cases where a source name stays the same but the type (or other info relevant to acking) changes.

Reply:
Thinking about it further, I see now none of this is actually needed. Since the new source is a distinct entity from the old source (the old source is shutdown), it *should* receive a new identifier on reload since the acknowledgements destined for the old source can no longer be delivered to the new one.

I am going to rewrite this to simply out any attempt to preserve the source ID, which will dramatically simplify this whole setup, mooting of most your questions.